crucible/src/Lang/Crucible/Simulator/BoundedExec.hs

```diff
@@ -202,48 +202,55 @@ boundedExecFeature getLoopBounds generateSideConditions =
                        }
 
  checkBackedge ::
-   IORef BoundedExecGlobal ->
+   IORef (Maybe BoundedExecGlobal) ->
    Some (BlockID blocks) ->
    BlockID blocks tgt_args ->
    SymGlobalState sym ->
    IO (SymGlobalState sym, Maybe Word64)
  checkBackedge gvRef (Some bid_curr) bid_tgt globals =
-   do gv <- readIORef gvRef
-      case fromMaybe [] (lookupGlobal gv globals) of
-        ( Right fbd : rest ) ->
-          do let id_curr = Ctx.indexVal (blockIDIndex bid_curr)
-             let id_tgt  = Ctx.indexVal (blockIDIndex bid_tgt)
-             let m = frameWtoMap fbd
-             case (Map.lookup id_curr m, Map.lookup id_tgt m) of
-               (Just (cx, _cd), Just (tx, td)) | tx <= cx ->
-                  do let cs       = frameBoundCounts fbd
-                     let (cs', q) = incrementBoundCount cs td
-                     let fbd'     = fbd{ frameBoundCounts = cs' }
-                     let globals' = insertGlobal gv (Right fbd' : rest) globals
-                     if q > frameBoundLimit fbd then
-                       return (globals', Just (frameBoundLimit fbd))
-                     else
-                       return (globals', Nothing)
-
-               _ -> return (globals, Nothing)
-        _ -> return (globals, Nothing)
+   do 
+    gv <- readIORef gvRef
+    case gv of 
+      Just gv -> 
```

Comment:
The vast majority of the rest of this function is in this `Just` branch. You could dedent it by instead writing
```haskell
let err = panic "checkBackedge" ["Global not initialized"]
gv <- fromMaybe err gv
-- everything in the `Just` case
```